### DISCUSSION TITLE
Operator doesn't support automatic renewal of certs

### DISCUSSION THREAD

**Discussion by ervikrant06** [Opster/opensearch-k8s-operator#394] [General]
As operator doesn't support automatic renewal of self-signed certs. What's the recommended practice to renew the self-signed certs by operator if cert expires?

**Comment by swoehrl-mw** [comment]
Hi @ervikrant06.
I haven't tested it but the following steps should work:
1. Manually delete the secrets containing the generated certificates (named `<clustername>-transport-cert` and `<clustername>-http-cert`).
2. Wait for the operator to regenerate the certificates (the operator will detect the missing secrets during its next reconcile run after 30 seconds and regenerate them).
3. Get the operator to do a rolling restart of all nodes, for example by adding a dummy change to the `additionalConfig`.

The cluster should stay operational during this as the still running nodes will work with the old certs and the CA stays the same (it is valid for 10 years) so nodes with old and new certs can validate each other.

I've also created a feature request (#399) to implement automatic renewal.

**Reply by ervikrant06** [reply]
Thanks @swoehrl-mw  for sharing the steps. It worked. 

- Only difference in <clustername>-transport-cert 

Before deleting the secret following keys were present.

~~~
        f:ca.crt: {}
        f:test-bootstrap-0.crt: {}
        f:test-bootstrap-0.key: {}
        f:test-masters-0.crt: {}
        f:test-masters-0.key: {}
        f:test-masters-1.crt: {}
        f:test-masters-1.key: {}
        f:test-masters-2.crt: {}
        f:test-masters-2.key: {}
        f:test-nodes-0.crt: {}
        f:test-nodes-0.key: {}
        f:test-nodes-1.crt: {}
        f:test-nodes-1.key: {}
~~~

After re-creation of secret by operator bootstrap key/crt was missing which prob we don't need. 

~~~
        f:ca.crt: {}
        f:test-masters-0.crt: {}
        f:test-masters-0.key: {}
        f:test-masters-1.crt: {}
        f:test-masters-1.key: {}
        f:test-masters-2.crt: {}
        f:test-masters-2.key: {}
        f:test-nodes-0.crt: {}
        f:test-nodes-0.key: {}
        f:test-nodes-1.crt: {}
        f:test-nodes-1.key: {}
~~~

I am hitting the [bug](https://github.com/Opster/opensearch-k8s-operator/issues/400) when the rolling restart of cluster nodes happen. Noticed this twice.

**Reply by swoehrl-mw** [reply]
Good to hear that it worked. Regarding the bootstrap cert: That is intential as it is only needed during initial cluster setup so it will not be recreated later on.
